File: 144/Solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public List<Integer> preorderTraversal(TreeNode root) {
        List<Integer> result = new LinkedList<Integer>();
        if (root == null)
            return result;
        Stack<TreeNode> stack = new Stack<TreeNode>();
        stack.push(root);
        while(!stack.empty()) {
            TreeNode visit = stack.pop();
            result.add(visit.val);
            if (visit.right != null)
                stack.push(visit.right);
            if (visit.left != null)
                stack.push(visit.left);
        }
        return result;
    }
}

File: 144/Solution.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<int> preorderTraversal(TreeNode* root) {
        vector<int> result;
        if (root == NULL)
            return result;
        stack<TreeNode *> stack;
        stack.push(root);
        while (!stack.empty()) {
            TreeNode* visit = stack.top();
            stack.pop();
            result.push_back(visit->val);
            if (visit -> right != NULL)
                stack.push(visit->right);
            if (visit -> left != NULL)
                stack.push(visit->left);
        }
        return result;
    }
};